src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1832,7 +1832,7 @@ static bool processOnChangeConditionForUpdateContext
     {
       for (unsigned int jx = 0; jx < attrL.size(); jx++)
       {
-        if (caP->name == attrL[jx] && !caP->skip)
+        if (caP->name == attrL[jx])
```

Comment:
Why did you need to remove the `&& !caP->skip` part? Could you elaborate on it, please?

Reply:
> Why did you need to remove the `&& !caP->skip` part? Could you elaborate on it, please?

As per my understanding this `caP->skip` field is used to mark deleted attributes that must not be included in the notification and this condition is used to add the attributes in the notification. So, I removed `!caP->skip` so that the deleted attributes included in the notification.

Reply:
Regardless of this new functionality, my doubt is if `&& !caP->skip` is really needed. Maybe it was needed in the past but some refactor in the code make it unneeded.

I'll cast and independent PR to check that.

Reply:
> I'll cast and independent PR to check that.

This one: PR https://github.com/telefonicaid/fiware-orion/pull/4201

Reply:
PR https://github.com/telefonicaid/fiware-orion/pull/4201 went ok, thus validating this change.

However, I'll investigate a little bit, to understand why this was there but now it is not needed :)

Reply:
Looking into this into more detail (see https://github.com/telefonicaid/fiware-orion/pull/4201#issuecomment-1233057677) I'm afraid that the modification is not valid.

This PR doesn't detect the problem due to in the moment the PR was created master didn't have the proper coverage in tests. Now that these test has been included (in PR https://github.com/telefonicaid/fiware-orion/pull/4204), if you upgrade  Anjali-NEC:issue2591 with master, we will be see some tests failing.